Sources/Workspace/Workspace.swift

```diff
@@ -4185,6 +4194,290 @@ extension Workspace.Location {
+            manifest: Manifest,
+            transformationMode: TransformationMode,
+            observabilityScope: ObservabilityScope,
+            callbackQueue: DispatchQueue,
+            completion: @escaping (Result<Manifest, Error>) -> Void
+        ) {
+            let sync = DispatchGroup()
+            let transformations = ThreadSafeKeyValueStore<PackageDependency, PackageIdentity>()
+            for dependency in manifest.dependencies {
+                if case .sourceControl(let settings) = dependency, case .remote(let url) = settings.location  {
+                    sync.enter()
+                    self.mapRegistryIdentity(url: url, observabilityScope: observabilityScope, callbackQueue: callbackQueue) { result in
+                        defer { sync.leave() }
+                        switch result {
+                        case .failure(let error):
+                            // do not raise error, only report it as warning
+                            observabilityScope.emit(warning: "failed querying registry identity for '\(url)': \(error)")
+                        case .success(.some(let identity)):
+                            transformations[dependency] = identity
+                        case .success(.none):
+                            // no identity found
+                            break
+                        }
+                    }
+                }
+            }
+
+            // update the manifest with the transformed dependencies
+            sync.notify(queue: .sharedConcurrent) {
+                do {
+                    let updatedManifest = try self.transformManifest(
+                        manifest: manifest,
+                        transformations: transformations.get(),
+                        transformationMode: transformationMode,
+                        observabilityScope: observabilityScope
+                    )
+                    completion(.success(updatedManifest))
+                }
+                catch {
+                    return completion(.failure(error))
+                }
+            }
+        }
+
+        private func transformManifest(
+            manifest: Manifest,
+            transformations: [PackageDependency: PackageIdentity],
+            transformationMode: TransformationMode,
+            observabilityScope: ObservabilityScope
+        ) throws -> Manifest {
+            var targetDependencyPackageNameTransformations = [String: String]()
+
+            var modifiedDependencies = [PackageDependency]()
+            for dependency in manifest.dependencies {
+                var modifiedDependency = dependency
+                if let registryIdentity = transformations[dependency] {
+                    guard case .sourceControl(let settings) = dependency, case .remote = settings.location else {
+                        // an implementation mistake
+                        throw InternalError("unexpected non-source-control dependency")
```

Comment:
Should the error contain the dependencies to make it easier to debug if this is every thrown?